File: Data Structures & Algorithms/binary-search/submission-3.java
class Solution {
    public int search(int[] nums, int target) {
        return binSearch(nums, 0, nums.length - 1, target);
    }

    int binSearch(int[] nums, int start, int end, int target) {
        // base
        if (start > end) return - 1;

        int mid = (start + end) / 2;
        
        if (target > nums[mid]) start = mid + 1;
        else if (target < nums[mid]) end = mid - 1;
        else return mid;

        return binSearch(nums, start, end, target);
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-3.go
func twoSum(nums []int, target int) []int {
    hmap := map[int]int{}

    for i, v := range nums {
        hmap[v] = i
    }

    for i, v := range nums {
        if j, ok := hmap[target - v]; ok && i != j {
            return []int{i, j}
        }
    }

    return []int{}
}
